bulldozer/merge.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bulldozer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/palantir/bulldozer/pull"
+)
+
+const MaxPullRequestPollCount = 5
+
+func MergePR(ctx context.Context, pullCtx pull.Context, client *github.Client, mergeConfig MergeConfig) error {
+	logger := zerolog.Ctx(ctx)
+
+	mergeOpts := &github.PullRequestOptions{}
```

Comment:
`PullRequestOptions` accepts a SHA value to make sure the PR is not updated before the merge call goes through. Is this something Bulldozer should pass to avoid race conditions?